### ISSUE TITLE
[QUESTION] How to merge the weights of sequence-parallel LayerNorm ?

### ISSUE THREAD

**Issue by Barber0** [NVIDIA/Megatron-LM#474]
I tried to pre-train a GPT-like language model with sequence-parallel. 

And I found that the LayerNorm weights in different ranks various.

`LayerNorm weights in Rank0:  tensor([0.5815, 0.6235, 0.5947,  ..., 0.4775, 0.4360, 0.4680],
       dtype=torch.float16)`

`LayerNorm weights in Rank1:  tensor([0.6885, 0.6792, 0.6880,  ..., 0.6987, 0.6108, 0.5713],
       dtype=torch.float16)`

If I want to merge these splited-weights into an intact set of weights that can run in single GPU, how to merge these LayerNorm weights?

**Comment by deepakn94** [comment]
Where are you printing these? We do an all-reduce already to "merge" the gradients across the different sequence-parallel replicas: https://github.com/NVIDIA/Megatron-LM/blob/805a3d5af69949e0266ec7c95a177bbfcb621331/megatron/core/pipeline_parallel/distrib_grad.py#L105

You shouldn't be seeing different LayerNorm weights.

**Comment by deepakn94** [comment]
Going to close this, please re-open if you are still running into this issue.
